src/main/scala/analysis.scala

```diff
@@ -165,30 +169,39 @@ case object analysis {
       override
       val sqsInitialTimeout =
         10.hours
+
+      override
+      val terminationConfig = TerminationConfig(
+        // if true loquat will terminate after solving all initial tasks
+        terminateAfterInitialDataMappings = true,
+        // if true loquat will terminate after errorQueue will contain more
+        // unique messages than threshold
+        errorsThreshold = Some(5),
+         // maximum time for processing one task
+        taskProcessingTimeout = None,
+        // maximum time for everything
+        globalTimeout = Some(6.hours)
+      )
     }
 
-    val defaultConfig =
+    def defaultConfig(id: String) =
       AnalysisConfig(
-        loquatName    = "data-analysis",
+        loquatName    = s"data-analysis-$id",
         logsS3Prefix  = S3Folder("miodx", "clonomap")/"analysis"/"log"/,
         managerConfig = DefaultManagerConfig
       )
 
     // worker
     //////////////////////////////////////////////////////////////////////////////
-    case object worker extends WorkerBundle(
-      analysisBundle,
-      defaultConfig
-    )
+    def createWorker(config: AnalysisConfig) = {
```

Comment:
Oh, wait a minute. This can be dangerous and is likely to fail. You shouldn't define `worker` object in a method, it has to have a stable reference, because it will be passed to the deployment script.

Reply:
Oops. I needed it to depend on `config`, as `config` depends on the loquat name. Maybe I could make the whole `case object analysis` a case class depending on an `id` and set the id from the webapp (?)

Reply:
I see that you have it worked around for the `manager` bundle by setting the `fullName` explicitly:

https://github.com/era7bio/asdfjkl/blob/3f20b5e5bc99f7408739346ce5f2ab454c3dcc90/src/main/scala/analysis.scala#L118-L125

So probably you can do the same in this `worker` object and it will be fine.